hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -43,16 +43,15 @@ String::String(const __FlashStringHelper *pstr)
 	*this = pstr;
 }
 
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if STRING_HAS_MOVE
 String::String(String &&rval)
+	: buffer(rval.buffer)
+	, capacity(rval.capacity)
+	, len(rval.len)
```

Comment:
I guess having `std::move` on all these assignments would allow move semantics / optimizations on each of the fields, but since we know these are just a pointer and two integers, it should not make any difference here. So I guess the code is good as the PR proposes it.